### ISSUE TITLE
Initial Theme demo

### ISSUE THREAD

**Issue by raysguy** [raysguy/HTPC-Manager#7]
@Hellowlol - just pushed a demo branch for you to have a look at.  It's the [demo/re-bootstrap] (https://github.com/raysguy/HTPC-Manager/tree/demo/re-bootstrap) branch.  It's up to date with your latest master2 commit.  Here's a quick rundown:

####Notes
- This is just a sample of three themes.  I've got lots more fully planned out (different colours / dark / light etc) and am just waiting to build them until I can fix the outstanding issues.
- The idea is to have a consistent UI so that making any changes to HTPC Manager won't break any individual themes.  If it works on one it should work on all.
- There will be a few dark themes to choose from but they'll take a bit more work than the others so I'm leaving them until after everything is fixed.
- Obsidian will be the replacement for a default theme.  Still have a few tweaks to make on that one with the button colouring but I just wanted to put something together to show you other than the coloured themes.

#### Changes/Enhancements
- Re-design of many UI elements and layout to give a more modern appearance and highlight content (e.g. more thumbnails visible in Plex module).
- Totally re-designed login page (which also changes nicely to match each theme).
- Lots of tweaks to make the experience more uniform - especially regarding dropdowns.
- Prettified labels (especially Sonarr and NZBGet modules).
- Changed alert boxes in Settings for more unified look and to make prettier!

#### New features
- Dropdown menus now expand simply by hovering over the button with the pointer.  More modern and fluid.  Clicking buttons still works so as to retain compatibility on touch devices.
- Replaced checkboxes with switches which I think looks nicer and also makes it easier to toggle options on touch devices.

#### Still to do
- Replace those goddam dropdown form selectors to get rid of the rounded corners!
- Implement slide in menu.

#### Known issues / need help
- The new switches to replace checkboxes work perfectly everywhere except the General settings page where they are too small.  Spent many hours trying to figure out what is different about that page but haven't been able to do anything about it so hopefully you'll spot what I've missed.
- Computer stats page - in the list of HDDs the "progress" bar is aligned to the top of the row and haven't been able to figure out how to get the vertical alignment to middle (can't figure out the right class to target).
- On mobile devices the HTPC Manager logo aligns to the top instead of middle, even though it's perfect in desktop view.
- Samsung Remote - still can't get that bloody search field centre aligned without screwing up mobile view!

**Comment by Hellowlol** [comment]
Looks amazing man!

**Comment by Hellowlol** [comment]
The reason the swichboxes does not work is because they are initiated before the page it ready. I would just add it here: https://github.com/raysguy/HTPC-Manager/blob/demo/re-bootstrap/interfaces/default/js/default.js#L6

**Comment by raysguy** [comment]
Glad you like them :)  Is it possible to force override the setting to Obsidian the first time the new themes are included? Or perhaps a way to remap existing settings so that if current config value is say 'amelia' or 'cosmo' those are directed to Obsidian?

I'll try to get a Kodi test server set up so I can check it out.  Could you take a couple of screenshots so I can see what you mean?

**Comment by Hellowlol** [comment]
We can remap it :).

I cant ss it as i just flashes. But the same happens on the plex module. Just scroll down fast or search for anything so you get one result up. I think we can fix this by adding overflow hidden. As for the modal error:

![image](https://cloud.githubusercontent.com/assets/1214134/8813493/3f27b5aa-3004-11e5-8ee9-32cdb2de0022.png)
We can fix that by adding 
````
.modal .modal-movieinfo {
  margin-left: 230px;
}
````

**Comment by Hellowlol** [comment]
Im very impressed with the work you have done sofar! When you ready (seems good enough for a merge after you have fixed the things i pointed out.) Open a pr and ill handle the remapping etc.

Any chance we can have a darker and a more rounded theme?

**Comment by raysguy** [comment]
Thanks, that's awesome, I'll get on the other themes.  There'll be a few dark options to choose from and I'll try to create a rounded version that looks good.  Will any of the fixes require changes to the theme files or just the htpc files?

I tried your suggestion for the switches and it worked like a charm thanks :)

**Comment by Hellowlol** [comment]
It's all good. The fixes is related to the modules nvm that I'll fix that later. If you have fixed the checkboxes open up a pr and I'll handle the migration to new themes before I merge and push it to github.

**Comment by raysguy** [comment]
Awesome.  As soon as I’ve built the rest of the themes I’ll send the PR :)

**Comment by Hellowlol** [comment]
I don't think you need to have every theme ready. I think the default is better then all the old ones combined. I think we can just merge it now, but it's up to you. :)

**Comment by Hellowlol** [comment]
http://forum.kodi.tv/showthread.php?tid=221715&pid=2059407#pid2059407

**Comment by raysguy** [comment]
Thanks for the shoutout, I appreciate it.  

How do you set kodi up properly?  I can’t seem to get it working to test.  Want to look at the now playing area - hadn’t realised it wasn’t the same as the Plex one so need to do a little work so I’m happy with it.

**Comment by Hellowlol** [comment]
Install kodi, add a folder, choose what content that folder is. enable webserver in the settings

**Comment by raysguy** [comment]
So I managed to get koi working and have changed the now playing section to look like it does in Plex, so that's much better now.

I've come across a weird issue though that I wanted to pick your brain about.  For no apparent reason the page-title bar has disappeared on the Kodi page. I see the buttons overlaying the main page content but the bar itself is missing. Inspecting the elements show it as having a height of 0px.  I've not made any changes to it myself and I've even loaded up old builds of htpc from over a month ago before I'd touched it and they exhibit the same behaviour.  Really puzzling me as I'm certain it was working yesterday and I can see in your screenshots it was showing correctly.  I've been over the koi files in detail and just can't figure it out.  I think the same is happening on the Headphones page too.  Could you check and see if you are experiencing the same?

Also I couldn't help noticing that nice non-rounded selector on your screenshot!  How did you manage that?  Nearly jumped for joy when I saw it!

**Comment by Hellowlol** [comment]
Yeah I can take a look. Can you push your latest changes ? as it seems fine here.
What selector are you talking about?

**Comment by raysguy** [comment]
Thanks, pushed it as demo/re-bootstrap2.

The tvtue2 selector in the page-title bar.

**Comment by raysguy** [comment]
Btw spotted another minor issue - in the Sonarr dash module when hovering over the 'i' button the episode description doesn't wrap and so isn't visible.  I traced the cause down to a 'whitespace: nowrap' element on the dash, but couldn't fix the tooltips without breaking the nowrap for everything else.

Another thing I've been tinkering with is the vnStat module.  It's always bugged me that the charts don't use the theme font.  However the only way I could find around this is to specify the font manually in the options at the end of vnstat.js.  That method isn't viable though as the font changes with the theme.  Can you think of any way to get around this and have the charts match the rest of the app?

**Comment by Hellowlol** [comment]
Yeah, I would have fetched the font type from some element you know are correct and store that as a js variable

**Comment by raysguy** [comment]
Thanks, I'm going to try and get that incorporated tonight.

**Comment by raysguy** [comment]
Just FYI @Hellowlol I've pushed a new branch (demo/re-bootstrap3) for you to take a look at as i've made quite a lot of progress today.  7 new themes added including Nostalgia which is the more rounded one you requested.  Also incorporated fixes for the errors you spotted earlier.

Tomorrow I'm going to working on a few dark themes and a really light one, plus a few rather ambitious themes I've planned but need to see if I can pull off!

**Comment by Hellowlol** [comment]
Sounds good! Btw, I noticed you removed the background image from now playing. Any chance we can keep it? I really like it. :D

**Comment by raysguy** [comment]
Yeah, I replied to your comment on that not long ago.  Do you know of a way to make the image monochrome and reduce the opacity so that it’s kinda like a ghost/watermark overlaying the black now playing well background?  Think that could work, I just have no clue how to do it! :)  I couldn’t leave it in as it was as it messed up the theme and I’m not sure how else I could achieve a consistent look across every possible background image.

On a related note, is it possible to get the fan art banner in the empty top-left space on the sonar_view page?  I was going to experiment with that but noticed you already had some commented out code in there relating to it.

**Comment by Hellowlol** [comment]
The o parameter in getthumb is for opacity. Pil supports monochrome. I can add support for that if you want to.

**Comment by Hellowlol** [comment]
Or client side with css http://davidwalsh.name/demo/css-filters.php

**Comment by raysguy** [comment]
Was just looking at something similar but everything seems to be webkit only.

**Comment by raysguy** [comment]
@Hellowlol I've been thinking about this some more and experimenting and I have an idea I think would look awesome if you're up for re-designing the now playing areal. Basically:
- If possible grab the banner art mentioned [here](http://kodi.wiki/view/Wide_banner_icons) instead of the fan art. Preferably the blank series ones.  I think they'd be really well suited for the background.
- We'd have a black bar running along the bottom (or underneath) the banner image containing the media controls, volume slider and then progress bar (similar to how most media players are set up).
- Don't know if it's possible but maybe add a "scrubber" to the progress bar for navigation.
- Get rid of the episode thumbnail on the left as a random still frame is rarely useful in identifying a show.  Instead the episode title and info could be moved to the top left of the well and underneath we could have the episode summary.  To avoid issues with visibility all that episode info could be in a semi-translucent container similar to the "Show details" container on the sonar_view page.
- If it works and looks good it could be rolled out to the Plex module also.

Let me know what you think.  If you could deal with the structure and the code grabbing the image I could handle the styling.  It's okay if you don't have time to look at it just yet because I can put the background-image code back into the kodi.js file but just comment it out and leave the background black for the short-term.

**Comment by Hellowlol** [comment]
Leave the code but comment it out. Ill see if i cant add support for monochome images. 
What is a scrubber? Can you make a mockup?

**Comment by raysguy** [comment]
This is a mock up of what I had in mind.  The scrubber is basically a slider like on the volume control.  You can drag it along the progress bar to move the video to a specific time.  Useful for quickly skipping part of a video or jumping back if you miss something.

![now playing demo](https://cloud.githubusercontent.com/assets/10825337/8886360/c8621f32-3260-11e5-967d-2815455324e9.png)

**Comment by Hellowlol** [comment]
Ah, you can press the progressbar i kodi btw ;)

**Comment by Hellowlol** [comment]
Added support for pil modes. See http://effbot.org/imagingbook/concepts.htm. You will need to add mode=yourmode to the url for it to work. I have only added it to the helper function and kodi, so i you want to use this in another module let me know.

**Comment by Hellowlol** [comment]
@raysguy Ill checkout new themes that you pushed today. I have been working on making the progressbar of plex module scrubbable (jump in the video) and adding support for banners in kodi.

**Comment by raysguy** [comment]
That's awesome.   I've been working on dark themes this weekend so I'll push an update with those for you to look at when I'm home.

**Comment by raysguy** [comment]
Just pushed new demo build 4 to my repo for you to look at.  Includes all the dark themes.  Took a bit longer to get them finished today as I came across a couple of bugs I'd missed which required some re-work.

**Comment by Hellowlol** [comment]
Np, I have been playing with the xmbc cache today, but I need to limit it. Arm it downloads every image. (I had 113862). Can wait to check it out. So if this is ok I can just merge it?

**Comment by raysguy** [comment]
Don't merge from the demo branch.  Still have some minor tweaks and polishing, plus some tidying (just using my wip files for demo so are duplicates etc).  Also have a few "feature" themes to finish.  Once those are done I'll create a proper release branch and create a PR.  Shouldn't take more than a day or two unless any major issues arise. :)

That's a lot of images, lol.  Will you be able to do the same with Plex too?

**Comment by Hellowlol** [comment]
No I don't think so, it's not worth it as the plex api is much harder work with. But I'll give it a try :)

**Comment by raysguy** [comment]
Awesome, I only use Plex as I need remote access for my girlfriend to access my stuff. Plus they have a great selection of apps across platforms.

**Comment by Hellowlol** [comment]
Ill do it for plex as well. Seems easy.

**Comment by Hellowlol** [comment]
The only issue i found was that the tab-content should have overflow hidden to remove the scoll bar for plex and kodi. I loved the dark themes, there was some really stuff that was really colorfull that i not so fond of, but im sure some will love them :D

**Comment by raysguy** [comment]
Yeah I fixed that the tab-content thing on my local build this morning.

Yeah some of them are not colours I would use myself but I wanted to have as wide a variety as possible so that there’s something for everyone.  Have to say I’m not normally too big on dark themes usually but sublime is one of my favourites so I’l probably use that quite a bit. :)
